standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -1928,9 +1928,8 @@ private void create_dataconnector_core(RawStore ms, final DataConnector connecto
     boolean success = false;
     Map<String, String> transactionalListenersResponses = Collections.emptyMap();
     try {
-      firePreEvent(new PreCreateDataConnectorEvent(connector, this));
-
       ms.openTransaction();
+      firePreEvent(new PreCreateDataConnectorEvent(connector, this));
```

Comment:
why the change?

Reply:
ditto, `firePreEvent` may throw exception and not call `openTransaction()`.

Reply:
What is the point? We open a txn, throw an exception in `firePreEvent`, and then abort txn?